### PR TITLE
Increased core dep from 3.1 to 3.2

### LINKED ISSUE #175: Error: Strict Standards: Declaration of Translatable...
I get this error after installing the translatable module in SS 3.1.6 and 3.1.8:

Strict Standards: Declaration of Translatable::augmentSQL() should be compatible with that of DataExtension::augmentSQL() in /home/sites/hesteric/www/translatable/code/model/Translatable.php on line 154

### PR DESCRIPTION
**Pull request by chillu**
3.2 still uses SQLQuery rather than SQLSelect, but changes the getWhere() return signature. Fixes #175

@tractorcow This will likely need a new release branch on the module (2.1), and the branch-alias on master changed to 2.2. That's assuming we don't create major module releases for new core compatibilities.

So don't merge this PR into 2.0, it'll break 3.1 core compat on that release line. Happy to rebase/resubmit the commit if you agree with the approach.

### CODE REVIEW

**Comment by tractorcow** [comment]
Is it possible at all to write this in such a way that this branch would support both 3.1 and 3.2? With Blog we were able to prevent having to create a ~3.2 version dependency; It would be nice if we could avoid this for translatable as well.

**Comment by chillu** [comment]
Hmm, I've tried to explain that in the first sentence of the PR. We could check for the `getWhere()` return signature to conditionally process the SQL statement for both 3.2 and 4.x compat though - what do you think?

File: code/model/Translatable.php [hunk 1]
```diff
@@ -612,13 +612,19 @@ function augmentSQL(SQLQuery &$query, DataQuery &$dataQuery = null) {
...
-			// call or through custom SQL filters)
-			&& !preg_match('/("|\'|`)Locale("|\'|`)/', implode(' ', $query->getWhere()))
 			//&& !$query->filtersOnFK()
 		)  {
-			$qry = sprintf('"%s"."Locale" = \'%s\'', $baseTable, Convert::raw2sql($locale));
-			$query->addWhere($qry); 
+			// Or we're already filtering by Lang (either from an earlier augmentSQL() 
+			// call or through custom SQL filters)
+			$filtersOnLocale = array_filter($query->getWhere(), function($predicates) {
+				foreach($predicates as $predicate => $params) {
```
Comment (tractorcow) [inline code/model/Translatable.php]: Ok, to answer the question and make a comment; the $predicates in this case would be one of;
- a string (3.1) containing the whole condition
- an array with a single item in `array('condition' => array('params'))` format.

Simple way to extract the string condition is.

```php
protected function filtersOnLocale($query) {
    foreach($query->getWhere() as $condition) {
        // Compat for 3.1/3.2 where syntax
        if(is_array($condition)) {
            reset($condition);
            $condition = key($condition);
        }
        if(preg_match('/("|\'|`)Locale("|\'|`)/', $condition)) return true;
    }
}
````

Comment (tractorcow) [inline code/model/Translatable.php]: I think fro 4.0 compat there's no point supporting the old syntax; The augmentSQL signature will take a SQLSelect object at this point and couldn't work with 3.x code anyway.

**Pull request closed without merging**